components/fhir-core/src/main/java/org/wso2/healthcare/codegen/tool/framework/fhir/core/FHIRSpecParser.java

```diff
@@ -202,21 +202,21 @@ public void parseIG(ToolConfig toolConfig, String igName, String igDirPath) {
                                 dataTypeDef.setKind(DefKind.fromCode(code));
                                 FHIRSpecificationData.getDataHolderInstance().addDataType(structureDefinition.getId(),
                                         dataTypeDef);
-                                OASGenerator oasGenerator = OASGenerator.getInstance();
-                                APIDefinition apiDefinition;
-                                if (fhirImplementationGuide.getApiDefinitions().containsKey(structureDefinition.getType())) {
-                                    apiDefinition = fhirImplementationGuide.
-                                            getApiDefinitions().get(structureDefinition.getType());
-                                } else {
-                                    apiDefinition = new APIDefinition();
-                                    apiDefinition.setResourceType(structureDefinition.getType());
-                                }
-                                apiDefinition.addSupportedProfile(structureDefinition.getUrl());
-                                apiDefinition.addSupportedIg(igName);
-                                apiDefinition.setOpenAPI(oasGenerator.generateResourceSchema(apiDefinition,
-                                        structureDefinition));
-                                fhirImplementationGuide.addApiDefinition(structureDefinition.getType(), apiDefinition);
                             }
+                            OASGenerator oasGenerator = OASGenerator.getInstance();
+                            APIDefinition apiDefinition;
+                            if (fhirImplementationGuide.getApiDefinitions().containsKey(structureDefinition.getType())) {
+                                apiDefinition = fhirImplementationGuide.
+                                        getApiDefinitions().get(structureDefinition.getType());
+                            } else {
+                                apiDefinition = new APIDefinition();
+                                apiDefinition.setResourceType(structureDefinition.getType());
+                            }
+                            apiDefinition.addSupportedProfile(structureDefinition.getUrl());
+                            apiDefinition.addSupportedIg(igName);
+                            apiDefinition.setOpenAPI(oasGenerator.generateResourceSchema(apiDefinition,
+                                    structureDefinition));
+                            fhirImplementationGuide.addApiDefinition(structureDefinition.getType(), apiDefinition);
```

Comment:
Let's move this logic only to "resource" type since OAS def for data types are not a valid scenario